uprotocol/client/usubscription/v3/inmemoryusubcriptionclient.py

```diff
@@ -70,32 +78,32 @@ def on_receive(self, message: UMessage) -> None:
                     pass
 
 
-class InMemorySubscriber(Subscriber):
+class InMemoryUSubscriptionClient(USubscriptionClient):
     """
-    The following is an in-memory implementation of the Subscriber interface that
-    wraps the UTransport for implementing the Subscriber-side of the pub/sub
-    messaging pattern to allow developers to subscribe and unsubscribe to topics.
-    This implementation uses the InMemoryRpcClient and SimpleNotifier interfaces
-    to invoke the subscription request message to the usubscription service, and register
-    to receive notifications for changes from the uSubscription service.
+    Implementation of USubscriptionClient that caches state information within the object
+    and used for single tenant applications (ex. in-vehicle). The implementation uses InMemoryRpcClient
+    that also stores RPC correlation information within the objects
     """
 
-    def __init__(self, transport: UTransport, rpc_client: RpcClient, notifier: Notifier):
+    def __init__(
+        self, transport: UTransport, rpc_client: Optional[RpcClient] = None, notifier: Optional[Notifier] = None
+    ):
         """
-        Creates a new subscriber for existing Communication Layer client implementations.
+        Creates a new USubscription client passing UTransport, CallOptions, and an implementation
+        of RpcClient and Notifier.
 
-        :param transport: The transport to use for sending the notifications
-        :param rpc_client: The RPC client to use for sending the RPC requests
-        :param notifier: The notifier to register notification listeners
+        :param transport: The transport to use for sending the notifications.
+        :param rpcClient: The RPC client to use for sending the RPC requests.
+        :param notifier: The notifier to use for registering the notification listener.
         """
         if not transport:
             raise ValueError(UTransport.TRANSPORT_NULL_ERROR)
         elif not isinstance(transport, UTransport):
```

Comment:
elif not needed. could have just been an 

**if not isinstance(transport, UTransport)**

Reply:
Fixed